### PR TITLE
Update golangci-lint to v1.55, fix lint errors

### PR DESCRIPTION
**Pull request by hairyhum**
## Change Overview

Currently golangci-lint is built with go version older than the go1.21 This can cause weird linter false-positives (e.g. https://github.com/kanisterio/kanister/actions/runs/7202258335/job/19620056020)

Updated version of golangci-lint in the build image Fixed linter warnings:
- deprecated rand.Seed
- non-sensical return in setupKopiaRepositoryServer

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [ ] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [ ] :sunflower: Feature
- [ ] :world_map: Documentation
- [x] :robot: Test

## Issues 

- fixes #issue-number

## Test Plan

- [ ] :muscle: Manual
- [x] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: pkg/kanctl/helper.go [hunk 1]
```diff
@@ -16,13 +16,12 @@ package kanctl
...
 import (
 	"math/rand"
-	"time"
 )
 
 const characters = "abcdefghijklmnopqrstuvwxyz0123456789"
 
 func init() {
-	rand.Seed(time.Now().UnixNano())
+
```
Comment (julio-lopez) [inline pkg/kanctl/helper.go:24]: delete the `init()` function altogether, since it is empty.

**Review by julio-lopez — APPROVED** [review]
+1: LG 👍
See nit comment.

**Comment by viveksinghggits** [comment]
Hi @hairyhum ,
I raised a PR https://github.com/kanisterio/kanister/pull/2545 to update the base image's version in our build image. And after that the lint has been failing with signal killed.
I think updating the the `golangci-lint` would be a good idea for  my PR as well. Let's merge this and then we can see why lint is complaining in my PR.

File: pkg/kanctl/helper.go (hunk 1 above)
Comment (julio-lopez) [reply pkg/kanctl/helper.go:24]: @hairyhum Please address this comment

Comment (julio-lopez) [reply pkg/kanctl/helper.go:24]: To be addressed in #2566

**Pull request merged**